src/lib.rs

```diff
@@ -1,9 +1,3221 @@
-pub mod shwild {
-    // primary module
+// src/lib.rs : Definition of the shwild Rust package
+
+// ///////////////////////////////////////////////
+// crate-level feature definitions
+
+
+// ///////////////////////////////////////////////
+// crate-level imports
+
+use std::{
+    error as std_error,
+    fmt as std_fmt,
+    result as std_result,
+};
+
+
+/// Represents parsing result.
+#[derive(Debug)]
+#[derive(PartialEq)]
+pub enum Error {
```

Comment:
Shouldn't you want a `Result<(), ParseError>` for this?  It seems unexpected that a type named `Error` could be `Ok`.

Reply:
Done